bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,939 @@
+package bitriseclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-io/go-utils/urlutil"
+	"github.com/bitrise-tools/go-xcode/certificateutil"
+	"github.com/bitrise-tools/go-xcode/profileutil"
+)
+
+const (
+	baseURL                      string = "https://api.bitrise.io/v0.1/"
+	appsEndPoint                 string = "/apps"
+	provisioningProfilesEndPoint string = "/provisioning-profiles"
+	certificatesEndPoint         string = "/build-certificates"
+)
+
+// Protocol ...
+type Protocol interface {
+	New(accessToken string) (apps []Application, err error)
+
+	SetSelectedAppSlug(selectedAppSlug string)
+
+	FetchUploadedProvisioningProfiles() ([]FetchUploadedProvisioningProfileListResponseData, error)
+
+	getUploadedProvisioningProfileDownloadURLBy(profileSlug string) (downloadURL string, err error)
+
+	downloadUploadedProvisioningProfile(downloadURL string) (content string, err error)
+
+	GetUploadedProvisioningProfileUUIDby(profileSlug string) (UUID string, err error)
+
+	RegisterProvisioningProfile(provisioningProfSize int64, profile profileutil.ProvisioningProfileInfoModel) (RegisterProvisioningProfileResponseData, error)
+
+	UploadProvisioningProfile(uploadURL string, uploadFileName string, outputDirPath string, exportFileName string) error
+
+	ConfirmProvisioningProfileUpload(profileSlug string, provUploadName string) error
+
+	FetchUploadedIdentities() ([]FetchUploadedIdentityListResponseData, error)
+
+	getUploadedIdentityDownloadURLBy(certificateSlug string) (downloadURL string, password string, err error)
+
+	downloadUploadedIdentity(downloadURL string) (content string, err error)
+
+	GetUploadedCertificatesSerialby(identitySlug string) (certificateSerialList []big.Int, err error)
+
+	RegisterIdentity(certificateSize int64) (RegisterIdentityResponseData, error)
+
+	UploadIdentity(uploadURL string, uploadFileName string, outputDirPath string, exportFileName string) error
+
+	ConfirmIdentityUpload(certificateSlug string, certificateUploadName string) error
+}
+
+// BitriseClient ...
+type BitriseClient struct {
+	accessToken     string
+	selectedAppSlug string
+}
+
+// New returns all the application of the user on Bitrise
+func (client *BitriseClient) New(accessToken string) (apps []Application, err error) {
```

Comment:
this should not be a function on an instance
func NewBitriseClient(...) ... {
}